app/src/androidTest/java/com/openclassrooms/realestatemanager/NetworkConnectionTest.java

```diff
@@ -0,0 +1,26 @@
+package com.openclassrooms.realestatemanager;
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule;
+import androidx.test.core.app.ApplicationProvider;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.filters.LargeTest;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static com.openclassrooms.realestatemanager.presentation.utils.Utils.isInternetAvailable;
+import static org.junit.Assert.assertTrue;
+
+@LargeTest
+@RunWith(AndroidJUnit4.class)
+public class NetworkConnectionTest {
+
+    @Rule
+    public InstantTaskExecutorRule instantTaskExecutorRule = new InstantTaskExecutorRule();
+
```

Comment:
I tough you'd enable an disable internet connection. Then test wether your Utils function behaves as expected.

Reply:
I've seen no way to enable disable connection for data and wifi programatically

Reply:
How about [https://stackoverflow.com/questions/56741122/unit-test-for-network-utility-class](url) ?

Reply:
i'me gonna test that

Reply:
<img width="504" alt="Capture" src="https://user-images.githubusercontent.com/60393680/107538335-24cb5700-6bc4-11eb-8390-01b5bb11fe22.PNG">


That's in what you've send me, if i understand well, after API17, every try to access phone data will fail cause protected.

I also couldn't disable wifi on test never pass :

assertNotEquals(networkInfo.getType(), ConnectivityManager.TYPE_WIFI);

neither more :

        assertFalse(wifiManager.isWifiEnabled());

Reply:
I was thinking about something like

`@Test
public void isWifiEnabled() {
    WifiManager wifiManager = (WifiManager) mContext.getSystemService(Context.WIFI_SERVICE);
    wifiManager.setWifiEnabled(false);
    wifiManager.disconnect();

    assertFalse(NetworkUtils.isConnectedWifi(mContext));
}`